### ISSUE TITLE
Hero Accessibility Improvements: Heading Levels

### ISSUE THREAD

**Issue by GCHQ-Developer-299** [mi6/ic-ui-kit#1687]
## Summary 
When 'heading' and 'subheading' are provided to Hero component as props, they do not render as semantic <hX> elements. 
Considering the implication of the prop names and the context that Hero components would be used in, these should be headings.

## 💬 Description 
Proposal: 
1. default 'heading' to an `<h2>` element and 'subheading' to an `<h3>`
2. add an optional 'heading-level' prop that defaults to "2" that sets the `<hX>` element of the 'heading' (with subheading being heading-level+1)

This will always meet WCAG standards, and if developers want further flexibility the component already supports Slots

## 💰 Use value 
Describe why this is important and how it will help users (either developers, designers or the end-users). Think about how it makes people's lives easier, more accessible or quicker to build stuff. 

## Additional info 
This requirement was defined in an accessibility testing session

**Comment by MI6-255** [comment]
Follow the same pattern as Page Header and have a "heading" prop etc, where you pass in both the text and the semantic tag e.g. <h2> etc

**Comment by MI6-255** [comment]
Need guidance for this

**Comment by GCHQ-Developer-299** [comment]
> Follow the same pattern as Page Header and have a "heading" prop etc, where you pass in both the text and the semantic tag e.g. h2 etc. This may already be implemented. The default should be a h2 on all sizes (not just small), the subheading may not be a heading as a default.

This isn't how the Page Header component functions currently, the heading prop is only used for text. I think the simple solution here is to default heading to `<h2>` and secondaryHeading to `<h3>` and then update the guidance to suggest using slots if devs want something else
